src/screens/Products/ProductList.js

```diff
@@ -0,0 +1,128 @@
+import { Wallpaper } from '@components'
+import React, { Component } from 'react'
+import { ActivityIndicator, Button, FlatList, StyleSheet, Text, TouchableOpacity, View } from 'react-native'
+import { ProfilePicture } from '../../components'
+import { withMappedNavigationProps } from 'react-navigation-props-mapper'
+
+class ProductList extends Component {
+  static navigationOptions = ({navigation}) => {
+    return { 
+      title: 'Product List',
+      headerRight: (
+        <Button
+          transparent
+          onPress={() => navigation.navigate('AddProduct', {
+            currentUID: navigation.getParam('currentUID'), 
+            onNavigateBack: navigation.getParam('onNavigateBack'),
+            renderArtisanDetail: navigation.state.params.onProductListBack
+          })}
+          title="Add"
+        />
+      )
+    }
+  }
+  
+  constructor(props) {
+    super(props) 
+
+    this.state = { 
+      showProductList: false,
+      currentProducts: this.props.currentProducts
+    }
+
+  }
+
+  componentDidMount() {
+    this.props.navigation.setParams({ onNavigateBack: this.handleOnNavigateBack })
+  }
+
+  handleOnNavigateBack = () => {
+    this.setState({
+      currentProducts: this.props.Products
+    })
+  };
+
+  _renderProductListItem = ({item, index}) => {
+
+    return (
+      <TouchableOpacity
+        testID={`listItem${index}`}
+        style={styles.productListView}
+        onPress={() => this.props.navigation.navigate('ProductDetail', {...item})}
+        key={item.key}
+      >
+        <ProfilePicture
+          source={{uri: item.mainPictureURL}}
+          style={styles.image}
+        />
+          
+        <View style={styles.nameItem}>
+          <Text style={styles.text}>{item.Title}</Text>
+          <Text style={styles.text}>{`Price: $${item.StandardPrice}`}</Text>
+          <Text style={styles.text}>{`Quantity: ${item.Quantity}`}</Text>
+        </View>
+      </TouchableOpacity>
+    )
+  }
+
+  _keyExtractor = (item) => item.productID
+
+  
```

Comment:
Probably don't need the extra spacing between these two methods. One spacing is good